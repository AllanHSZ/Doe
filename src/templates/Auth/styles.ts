import styled, { css } from 'styled-components'
import media from 'styled-media-query'

import { FormLink } from 'components/Form'
import { Footer } from 'components/Footer/styles'

import { AuthProps } from '.'

type MainProps = Pick<AuthProps, 'maxHeight'>

export const Wrapper = styled.div`
  display: flex;
  height: 100vh;
  width: 100vw;
`

export const Decoration = styled.section`
  ${({ theme }) => css`
    flex: 1;
    display: none;
    align-items: center;
    justify-content: center;
    background-color: ${theme.colors.lightGray};
    max-width: 60vw;
    padding: 10vh 10vw;

    @media screen and (min-width: 1000px) {
      display: flex;
    }
  `}
`

export const Content = styled.section`
  ${({ theme }) => css`
    flex: 1;
    display: grid;
    gap: ${theme.spacings.xsmall};
    grid-template-rows: auto auto ${theme.font.sizes.medium};
    padding: ${theme.spacings.small} ${theme.spacings.large};
    overflow: auto;

    ${Footer} {
      margin: auto 0;
    }

    @media screen and (max-width: 1169px) {
      padding: ${theme.spacings.small};

      ${Footer} {
        margin: 0 auto;
      }

      ${FormLink} {
        flex-direction: column;
        margin: auto;
      }
    }
  `}
`

export const Header = styled.header`
  display: flex;
  align-self: flex-end;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: min(100%, 24rem);

  ${media.greaterThan('large')`
    align-items: flex-start;
  `}
`

export const Heading = styled.h2`
  ${({ theme }) => css`
    max-width: 200px;
    text-align: center;
    margin: ${theme.spacings.xxsmall} 0 0;

    p {
      text-align: center;
    }

    ${media.greaterThan('large')`
      text-align: start;
    `}
  `}
`

export const SubTitle = styled.h5`
  ${({ theme }) => css`
    font-weight: ${theme.font.normal};
  `}
`

export const Main = styled.main<MainProps>`
  ${({ theme, maxHeight }) => css`
    display: flex;
    margin: 0 auto;
    flex-direction: column;
    gap: ${theme.spacings.xsmall};
    justify-content: space-between;
    height: min(100%, ${maxHeight});
    align-items: center;
    max-width: 36rem;
    width: 100%;

    ${FormLink} {
      margin-top: auto;
      margin-bottom: auto;
    }

    ${media.greaterThan('large')`
      align-items: flex-start;
      margin: 0 auto 0 0;
    `}
  `}
`
