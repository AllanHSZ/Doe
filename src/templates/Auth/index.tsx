import Link from 'next/link'

import useMedia from 'hooks/useMedia'
import { FormLink } from 'components/Form'
import Logo from 'components/Logo'
import Footer from 'components/Footer'

import * as S from './styles'

export type AuthProps = {
  title: string
  subtitle?: string
  redirectText: string
  redirectLink: string
  redirectLinkText: string
  formHeight: number
  children: React.ReactNode
}

const Auth = ({
  title,
  children,
  subtitle,
  redirectText,
  redirectLink,
  redirectLinkText,
  formHeight
}: AuthProps) => {
  const greaterThenMedium = useMedia('(min-width: 1000px)')

  return (
    <S.Wrapper>
      {greaterThenMedium && (
        <S.Decoration>
          <svg
            viewBox="0 0 302 282"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g clipPath="url(#authDecor)">
              <path
                d="M193.067.009c-16.04-.401-34.458 12.994-75 48-72.5 7-120.109 33.016-118 81 2 45.5 68.5 58 112.5 89 28 25.5 43.354 57.867 80.5 52.5 131.5-19 128-252.5 0-270.5Z"
                fill="#E9F2FF"
              />
              <path
                d="M111.567 1.509c26.5.5 30.125 18 13.5 28.5-9.5 6-16.735 7.905-30.5 11-28 1-32-18-10.5-31 9.55-5.107 15.468-7.124 27.5-8.5Z"
                fill="#E9F2FF"
              />
              <path
                d="M114.067 63.509c16-2.872 57 8.5 57 8.5-.166 10.48-13.445 11.719-54 8l16 60h-54s-31.5-2.5-32.5-33.5c-1-31.02 43.578-38.707 67.5-43ZM259.067 157.009l-43.5-25.5-15.5 1c-16.614 18.819-29.254 23.377-56.5 23-8 0-6.5 12 0 12h21.5l-3.5 1c-9 0-8 12.5 0 12.5l71.5 1c14.686-3.999 20.763-8.985 26-25Z"
                fill="#2E78D3"
              />
              <path
                d="M189.067 75.508c-17-.966-26.5 15-26.5 15s-9.5-15-28-15c-15.486 0-26 14.014-26 29.5 0 37.001 54 64 54 64s53-25.499 53-64c0-15.486-9.5-28.533-26.5-29.5Z"
                fill="#04D361"
                stroke="#04D361"
              />
              <path
                d="M139.067 104.009c-15.341-23.295-26-35.5-50-35.5-22.233 0-42.5 18-42.5 39 0 32 48 19.5 46-18.5 12.229.975 17.064 3.267 22 10.5 7.5 9.5 17.725 10.872 24.5 4.5ZM221.067 112.009c-24.449-3.069-42.281 14.122-54.5 33.5 4.481 5.941 8.485 6.964 20 2 8.407-11.191 13.892-16.01 28.5-15.5-6.5 29.5 40 52 45.5 18-2.502-21.891-17.3-35.214-39.5-38Z"
                fill="#6492F2"
              />
            </g>
            <defs>
              <clipPath id="authDecor">
                <path fill="#fff" d="M0 0h290v271H0z" />
              </clipPath>
            </defs>
          </svg>
        </S.Decoration>
      )}

      <S.Content>
        <S.HeaderSection>
          <Logo withTitle={greaterThenMedium} />
          <div>
            <S.Heading>{title}</S.Heading>
            {!!subtitle && <p>{subtitle}</p>}
          </div>
        </S.HeaderSection>

        <S.FormWrapper formHeight={formHeight}>
          {children}
          <FormLink>
            {redirectText}{' '}
            <Link href={redirectLink}>
              <a>{redirectLinkText}</a>
            </Link>
          </FormLink>
        </S.FormWrapper>
      </S.Content>
      <Footer />
    </S.Wrapper>
  )
}

export default Auth
