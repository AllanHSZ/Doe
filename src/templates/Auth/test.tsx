import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Auth from '.'

const Props = {
  title: 'Auth simple title',
  redirectText: 'Redirect text',
  redirectLink: '/link',
  redirectLinkText: 'Redirect link text',
  maxHeight: '50vh'
}

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn()
  }))
})

describe('<Auth />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(
      <Auth {...Props}>
        <p data-testid="authChildren">Children</p>
      </Auth>
    )

    expect(
      screen.getByRole('heading', { name: /Auth simple title/i })
    ).toBeInTheDocument()

    expect(screen.getByText(/Redirect text/i)).toBeInTheDocument()

    expect(
      screen.getByRole('link', { name: /Redirect link text/i })
    ).toHaveAttribute('href', '/link')

    expect(screen.getByTestId('authChildren')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the subtitle', () => {
    renderWithTheme(
      <Auth {...Props} subtitle="A simple subtitle">
        <p data-testid="authChildren">Children</p>
      </Auth>
    )

    expect(
      screen.getByRole('heading', { name: /A simple subtitle/i })
    ).toBeInTheDocument()

    expect(screen.getByTestId('authChildren')).toBeInTheDocument()
  })
})
