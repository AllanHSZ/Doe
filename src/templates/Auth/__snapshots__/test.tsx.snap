// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Auth /> should render correctly 1`] = `
.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.8rem;
  font-size: 1.4rem;
  color: #394338;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c8 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  color: #04D361;
  border-bottom: 0.2rem solid #04D361;
  -webkit-transition: color,border,0.1s ease-in-out;
  transition: color,border,0.1s ease-in-out;
}

.c8 a:hover {
  color: #037938;
  border-bottom: 0.1rem solid #037938;
}

.c4 {
  width: 8rem;
  height: 7rem;
}

.c10 {
  color: #394338;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c11 {
  font-size: 1.2rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
  width: 100vw;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #F2F3F5;
  max-width: 60vw;
  padding: 10vh 10vw;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: grid;
  gap: 1.6rem;
  grid-template-rows: auto auto 1.6rem;
  padding: 2.4rem 4.0rem;
  overflow: auto;
}

.c2 .c9 {
  margin: auto 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: min(100%,24rem);
}

.c5 {
  max-width: 200px;
  text-align: center;
  margin: 0.8rem 0 0;
}

.c5 p {
  text-align: center;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.6rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: min(100%,50vh);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 36rem;
  width: 100%;
}

.c6 .c7 {
  margin-top: auto;
  margin-bottom: auto;
}

@media screen and (min-width:1000px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media screen and (max-width:1169px) {
  .c2 {
    padding: 2.4rem;
  }

  .c2 .c9 {
    margin: 0 auto;
  }

  .c2 .c7 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin: auto;
  }
}

@media (min-width:1170px) {
  .c3 {
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }
}

@media (min-width:1170px) {
  .c5 {
    text-align: start;
  }
}

@media (min-width:1170px) {
  .c6 {
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
    margin: 0 auto 0 0;
  }
}

<div
  class="c0"
>
  <section
    class="c1"
  >
    <svg
      aria-label="Duas mãos segurando um coração"
      fill="none"
      viewBox="0 0 302 282"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#authDecor)"
      >
        <path
          d="M193.067.009c-16.04-.401-34.458 12.994-75 48-72.5 7-120.109 33.016-118 81 2 45.5 68.5 58 112.5 89 28 25.5 43.354 57.867 80.5 52.5 131.5-19 128-252.5 0-270.5Z"
          fill="#E9F2FF"
        />
        <path
          d="M111.567 1.509c26.5.5 30.125 18 13.5 28.5-9.5 6-16.735 7.905-30.5 11-28 1-32-18-10.5-31 9.55-5.107 15.468-7.124 27.5-8.5Z"
          fill="#E9F2FF"
        />
        <path
          d="M114.067 63.509c16-2.872 57 8.5 57 8.5-.166 10.48-13.445 11.719-54 8l16 60h-54s-31.5-2.5-32.5-33.5c-1-31.02 43.578-38.707 67.5-43ZM259.067 157.009l-43.5-25.5-15.5 1c-16.614 18.819-29.254 23.377-56.5 23-8 0-6.5 12 0 12h21.5l-3.5 1c-9 0-8 12.5 0 12.5l71.5 1c14.686-3.999 20.763-8.985 26-25Z"
          fill="#2E78D3"
        />
        <path
          d="M189.067 75.508c-17-.966-26.5 15-26.5 15s-9.5-15-28-15c-15.486 0-26 14.014-26 29.5 0 37.001 54 64 54 64s53-25.499 53-64c0-15.486-9.5-28.533-26.5-29.5Z"
          fill="#04D361"
          stroke="#04D361"
        />
        <path
          d="M139.067 104.009c-15.341-23.295-26-35.5-50-35.5-22.233 0-42.5 18-42.5 39 0 32 48 19.5 46-18.5 12.229.975 17.064 3.267 22 10.5 7.5 9.5 17.725 10.872 24.5 4.5ZM221.067 112.009c-24.449-3.069-42.281 14.122-54.5 33.5 4.481 5.941 8.485 6.964 20 2 8.407-11.191 13.892-16.01 28.5-15.5-6.5 29.5 40 52 45.5 18-2.502-21.891-17.3-35.214-39.5-38Z"
          fill="#6492F2"
        />
      </g>
      <defs>
        <clippath
          id="authDecor"
        >
          <path
            d="M0 0h290v271H0z"
            fill="#fff"
          />
        </clippath>
      </defs>
    </svg>
  </section>
  <section
    class="c2"
  >
    <header
      class="c3"
    >
      <div
        class="c4"
      >
        <svg
          aria-label="Doe"
          fill="none"
          height="100%"
          role="img"
          viewBox="0 0 100 90"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#a)"
            filter="url(#b)"
          >
            <path
              d="M91.977 31.877c-1.5-21.814-25.121-30.088-40.495-17.3l9 8.65c7.498 10.906-2.25 19.556-13.499 13.539L28.61 55.571c-3.374 4.137.375 10.154 6 7.146l20.997-20.31 2.25 1.88-20.623 21.438c-2.624 4.513 1.875 9.026 7.124 6.018l20.248-20.31 2.624 1.881-20.622 21.061c-3.374 4.513 2.25 10.155 6.75 6.394l32.245-31.968c7.124-9.402 6.374-16.924 6.374-16.924Z"
              fill="#04D361"
            />
            <path
              d="m44.733 34.51-6.374-7.146-18.748 18.804c-4.499 4.137 1.5 12.411 7.125 6.77l17.997-18.429Z"
              fill="#04D361"
            />
            <path
              d="M47.733 14.2c-4.124-3.384-15.748-11.659-30.746-.752C1.14 24.972 9.863 45.792 15.862 50.305c-.75-2.256 1.5-5.641 2.625-6.77l17.997-18.428-5.999-5.641 1.875-1.505 15.748 15.796c8.249 8.274 16.498-2.632 9.749-9.402 0 0-5.876-6.669-10.124-10.155Z"
              fill="#04D361"
            />
            <path
              d="M91.977 31.877c-1.5-21.814-25.121-30.088-40.495-17.3l9 8.65c7.498 10.906-2.25 19.556-13.499 13.539L28.61 55.571c-3.374 4.137.375 10.154 6 7.146l20.997-20.31 2.25 1.88-20.623 21.438c-2.624 4.513 1.875 9.026 7.124 6.018l20.248-20.31 2.624 1.881-20.622 21.061c-3.374 4.513 2.25 10.155 6.75 6.394l32.245-31.968c7.124-9.402 6.374-16.924 6.374-16.924Z"
              stroke="#000"
              stroke-width=".5"
            />
            <path
              d="m44.733 34.51-6.374-7.146-18.748 18.804c-4.499 4.137 1.5 12.411 7.125 6.77l17.997-18.429Z"
              stroke="#000"
              stroke-width=".5"
            />
            <path
              d="M47.733 14.2c-4.124-3.384-15.748-11.659-30.746-.752C1.14 24.972 9.863 45.792 15.862 50.305c-.75-2.256 1.5-5.641 2.625-6.77l17.997-18.428-5.999-5.641 1.875-1.505 15.748 15.796c8.249 8.274 16.498-2.632 9.749-9.402 0 0-5.876-6.669-10.124-10.155Z"
              stroke="#000"
              stroke-width=".5"
            />
          </g>
          <defs>
            <clippath
              id="a"
            >
              <path
                d="M0 0h100v89H0z"
                fill="#fff"
              />
            </clippath>
            <filter
              color-interpolation-filters="sRGB"
              filterUnits="userSpaceOnUse"
              height="82.5"
              id="b"
              width="92.5"
              x="3.75"
              y="7.75"
            >
              <feflood
                flood-opacity="0"
                result="BackgroundImageFix"
              />
              <fecolormatrix
                in="SourceAlpha"
                result="hardAlpha"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <feoffset
                dy="4"
              />
              <fegaussianblur
                stdDeviation="2"
              />
              <fecolormatrix
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
              />
              <feblend
                in2="BackgroundImageFix"
                result="effect1_dropShadow_333:569"
              />
              <feblend
                in="SourceGraphic"
                in2="effect1_dropShadow_333:569"
                result="shape"
              />
              <fecolormatrix
                in="SourceAlpha"
                result="hardAlpha"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <feoffset />
              <fegaussianblur
                stdDeviation="5"
              />
              <fecomposite
                in2="hardAlpha"
                k2="-1"
                k3="1"
                operator="arithmetic"
              />
              <fecolormatrix
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
              />
              <feblend
                in2="shape"
                result="effect2_innerShadow_333:569"
              />
            </filter>
          </defs>
        </svg>
      </div>
      <h2
        class="c5"
      >
        Auth simple title
      </h2>
    </header>
    <main
      class="c6"
    >
      <p
        data-testid="authChildren"
      >
        Children
      </p>
      <div
        class="c7 c8"
      >
        <p>
          Redirect text
        </p>
        <a
          href="/link"
        >
          Redirect link text
        </a>
      </div>
    </main>
    <footer
      class="c9 c10"
    >
      <p
        class="c11"
      >
        Doe © 2021, Nenhum direito reservado.
      </p>
    </footer>
  </section>
</div>
`
